examples/property_prediction/pretrain_gnns/chem/pretrain_supervised.py

```diff
@@ -0,0 +1,144 @@
+    parser.add_argument('--JK', type=str, default="last",
+                        help='how the node features across layers are combined. last, sum, max or concat')
+    parser.add_argument('--dataset', type=str, default='chembl_filtered',
+                        help='root directory of dataset. For now, only classification.')
+    parser.add_argument('--gnn_type', type=str, default="gin")
+    parser.add_argument('--input_model_file', type=str, default='', help='filename to read the model (if there is any)')
+    parser.add_argument('--output_model_file', type=str, default='', help='filename to output the pre-trained model')
+    parser.add_argument('--seed', type=int, default=0, help="Seed.")
+    parser.add_argument('--num_workers', type=int, default=8, help='number of workers for dataset loading')
+    args = parser.parse_args()
+    print(args)
+
+    torch.manual_seed(args.seed)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed_all(args.seed)
+
+    device = torch.device("cuda:" + str(args.device)) if torch.cuda.is_available() else torch.device("cpu")
+
+    model = GINPredictor(num_node_emb_list=[119, 4],
+                         num_edge_emb_list=[6, 3],
+                         num_layers=args.num_layer,
+                         emb_dim=args.emb_dim,
+                         JK=args.JK,
+                         dropout=args.dropout_ratio,
+                         readout=args.graph_pooling,
+                         n_tasks=1310)
+
+    if not args.input_model_file == "":
+        model.load_state_dict(torch.load(args.input_model_file + ".pth"))
+    model.to(device)
+
+    if args.zinc_standard_agent is not 'chembl_filtered':
+        raise ValueError('Dataset should be chembl_filtered.')
+    with open('./supervised_chembl_rev.pkl', 'rb') as f:
+        data = pickle.load(f)
+
+    atom_featurizer = PretrainAtomFeaturizer()
+    bond_featurizer = PretrainBondFeaturizer()
+    dataset = PretrainSupervisedMoleculeCSVDataset(data=data,
+                                                   smiles_to_graph=partial(smiles_to_bigraph, add_self_loop=True),
+                                                   node_featurizer=atom_featurizer,
+                                                   edge_featurizer=bond_featurizer)
+
+    train_dataloader = DataLoader(dataset=dataset,
+                                  batch_size=args.batch_size,
+                                  shuffle=True,
+                                  num_workers=args.num_workers,
+                                  collate_fn=collate)
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr, weight_decay=args.decay)
+    criterion = nn.BCEWithLogitsLoss(reduction='none')
+
+    train(args, model, train_dataloader, optimizer, criterion, device)
+
+    if not args.output_model_file == "":
+        torch.save(model.state_dict(), args.output_model_file + ".pth")
```

Comment:
ditto for `args.output_model_file + ".pth"`

Reply:
done